### DISCUSSION TITLE
Optional package dependencies?

### DISCUSSION THREAD

**Discussion by bilderbuchi** [JuliaPlots/Makie.jl#2683] [General]
Is using optional dependencies feasible for Makie, e.g. using the new [extensions](https://pkgdocs.julialang.org/dev/creating-packages/#Conditional-loading-of-code-in-packages-(Extensions)) feature or `targets` in Project.toml?

I'm curious, because I'm currently prototyping a GUI application using Makie (and Gtk) that I need to PackageCompile.
During playing around with that process, I noticed that the result is huge (another story I guess). I noticed that I'm pulling in an `FFMPEG` dependency via `Makie`. Also, the (largeish) MKL is pulled lazily in via `FFTW`<-`KernelDensity`<-`Makie`. 

I'm not including video playback features, so I thought it should be possible to not install FFMPEG. However, this is `use`d in the `Makie` module, which is of course centrally used everywhere.
Similar with the MKL - it would be great to avoid that if I'm not using violin or density plots.

I'm not asking for this to be implemented, because it's probably a lot of work, and at this point (a 1.25 GB packagecompiler result for HDF5+GLMakie+Gtk4+GtkMakie is not encouraging) I'm not even sure if I won't be going back to Python/pyinstaller to attempt this - it might be too early for Julia for this kind of usecase. :-/

**Comment by ndgnuh** [comment]
Hi, I'm having the same problem and I need to reduce the package size. Have you figured out a workaround for this?

**Reply by bilderbuchi** [reply]
No, unfortunately, I dropped considering Julia for this use case.

**Reply by ndgnuh** [reply]
What did you use instead, I'm open for suggestion?

**Reply by bilderbuchi** [reply]
Dropped the project, it was just an exploratory prototype. Would probably use C# if I had more experience in that, skill-wise I'd prefer Python, but there the situation is similar to Julia. C++ would also be an option.

**Comment by SimonDanisch** [comment]
I wish we had tools for that in PackageCompiler...
The new julia --trim option sadly doesn't address this, but maybe can in the future...

**Comment by aplavin** [comment]
Re weakdeps: would be great to utilize them, see https://github.com/MakieOrg/Makie.jl/issues/3953 and https://github.com/MakieOrg/Makie.jl/discussions/4277.
